tests/e2e/utils/data_fetcher/logs.py

```diff
@@ -0,0 +1,39 @@
+import json
+from functools import lru_cache
+from typing import List, Optional, Union
+
+from mypy_boto3_cloudwatch import CloudWatchClient
+from pydantic import BaseModel
+from retry import retry
+
+
+class Log(BaseModel):
+    level: str
+    location: str
+    message: Union[dict, str]
+    timestamp: str
+    service: str
+    cold_start: Optional[bool]
+    function_name: Optional[str]
+    function_memory_size: Optional[str]
+    function_arn: Optional[str]
+    function_request_id: Optional[str]
+    xray_trace_id: Optional[str]
+    extra_info: Optional[str]
+
+
+@lru_cache(maxsize=10, typed=False)
+@retry(ValueError, delay=1, jitter=1, tries=20)
+def get_logs(lambda_function_name: str, log_client: CloudWatchClient, start_time: int, **kwargs: dict) -> List[Log]:
+    response = log_client.filter_log_events(logGroupName=f"/aws/lambda/{lambda_function_name}", startTime=start_time)
+    if not response["events"]:
+        raise ValueError("Empty response from Cloudwatch Logs. Repeating...")
+    filtered_logs = []
+    for event in response["events"]:
+        try:
+            message = Log(**json.loads(event["message"]))
```

Comment:
I think we need to decide whether to use Poetry models in data_fetchers/data_builder for the sake of consistency. Here I used it, but I can see that in other places you use typdef taken from mypy_boto3 directly and return simple objects